mephisto/abstractions/blueprints/parlai_chat/parlai_chat_task_runner.py

```diff
@@ -263,6 +270,9 @@ def run_unit(self, unit: "Unit", agent: "Agent") -> None:
         while not world.episode_done() and unit.db_id in self.running_units:
             world.parley()
 
+        # Ensure agent can submit after completion
+        agent.observe({"state": {"task_done": True}})
```

Comment:
this seems more concise: `agent.observe({"task_done": True})`

or perhaps even this: `agent.update_state({"task_done": True})`

Reply:
Maybe this is related to my comments [here](https://github.com/facebookresearch/Mephisto/pull/660/files#r806126451) and [here](https://github.com/facebookresearch/Mephisto/pull/660/files#r806108770) in #660.

Instead of first-classing a `state` key, at the library level as convention, what if we didn't? I'm happy keeping the `state` key for bootstrap-chat, but maybe we should just consider state updates as a special class of `messages`?

Reply:
>  instead of first-classing a state key, at the mephisto-task library level as convention, what if we didn't? I'm happy keeping the state key for bootstrap-chat, but what if we just considered state updates as a special class of messages?

I've been trying to figure this one out as well. The one thing that seems to *require* state is what I considered a common pattern of [requesting an act](https://github.com/facebookresearch/Mephisto/blob/ddc5095a55f51e1664cd325dd1e89bf514fb3c34/mephisto/operations/client_io_handler.py#L326-L331) when [acting and none is available](https://github.com/facebookresearch/Mephisto/blob/ddc5095a55f51e1664cd325dd1e89bf514fb3c34/mephisto/data_model/agent.py#L308-L314). If we remove state entirely, I'm not sure what we change for this - do we require `Blueprint`s to identify and recreate this pattern on a case-by-case basis?

Reply:
So in the end I'm using `task_data` as most other messages sent by parlai agents are `{'text': ..., 'task_data': ...}`, so it makes sense to maintain this design pattern rather than introducing one (considering state is task specific now!)

Reply:
gotcha, makes sense!